builtin/credential/approle/path_role.go

```diff
@@ -1726,6 +1739,15 @@ func (b *backend) pathRoleCreateUpdate(ctx context.Context, req *logical.Request
 		resp.AddWarning("token_max_ttl is greater than the backend mount's maximum TTL value; issued tokens' max TTL value will be truncated")
 	}
 
+	if txn != nil {
+		if err := txn.Commit(ctx); err != nil {
+			return nil, fmt.Errorf("failed to commit transaction during renewal: %w", err)
+		}
+	}
+
+	// Restore the original storage
+	req.Storage = originalStorage
+
 	// Store the entry.
 	return resp, b.setRoleEntry(ctx, req.Storage, role.name, role, previousRoleID)
```

Comment:
@v0nNemizez This won't work well: the last storage write occurs outside of the transaction. I think you want to hoist this up above transaction commit.